Lib/unittest/mock.py

```diff
@@ -817,6 +817,12 @@ def _get_call_signature_from_name(self, name):
             if child is None or isinstance(child, _SpecState):
                 break
             else:
+                # If an autospecced object is attached using attach_mock the
+                # child would be a function with mock object as attribute from
+                # which signature has to be derived. If there is no signature
```

Comment:
You might want to review this comment (maybe remove it as well and include it in the commit message).

Reply:
Thanks for catching this. I prefer to keep it here with the part about fallback updated since it will be good to know why extract_mock is needed. I will add this in the comment over this change.